lib/taglet/tag_as.ex

```diff
@@ -45,4 +49,32 @@ defmodule Taglet.TagAs do
       end
     end
   end
+
+  defmacro __before_compile__(_env) do
+
+    quote do
+      @contexts
+      |> Enum.each(fn(context) ->
```

Comment:
In this macro I've put only functions that require the __MODULE__ compiled before, It's possible add it all together but I think it's better in this way, just to know the reason.